evap/results/views.py

```diff
@@ -81,7 +85,11 @@ def course_detail(request, semester_id, course_id):
         results = []
         for result in section.results:
             if isinstance(result, TextResult):
-                answers = [answer for answer in result.answers if user_can_see_text_answer(request.user, represented_users, answer, public_view)]
+                if view == 'filter':
+                    answers = [answer for answer in result.answers if user_can_see_text_answer(UserProfile.objects.get(id=contributor_id), [], answer, view)]
+                else:
+                    answers = [answer for answer in result.answers if user_can_see_text_answer(request.user, represented_users, answer, view)]
```

Comment:
above is some code collecting the represented users. if you set that to `[]` in case of the filtered view, then you don't need the if else in here